dataset/bin.cpp

```diff
@@ -452,30 +511,25 @@ DataArray groupby_concat_bins(const DataArrayConstView &array,
   const auto masked = hide_masked();
   TargetBins<DataArrayConstView> target_bins(masked, builder.dims());
   builder.build(*target_bins, array.coords());
-  return add_metadata(
-      bin<DataArrayConstView>(masked, *target_bins, builder.dims()),
-      array.coords(), array.masks(), array.attrs(), builder.edges(),
-      builder.groups(), {reductionDim});
+  return add_metadata(bin<DataArrayConstView>(masked, *target_bins, builder),
+                      array.coords(), array.masks(), array.attrs(),
+                      builder.edges(), builder.groups(), {reductionDim});
 }
 
 namespace {
 void validate_bin_args(const std::vector<VariableConstView> &edges,
                        const std::vector<VariableConstView> &groups) {
-  if (edges.empty()) {
-    if (groups.empty()) {
-      throw except::BucketError(
-          "Arguments 'edges' and 'groups' of scipp.bin are "
-          "both empty. At least one must be set.");
-    }
-  } else {
-    for (std::size_t i = 0; i < edges.size(); ++i) {
-      if (const auto &shape = edges[i].dims().shape();
-          std::any_of(shape.begin(), shape.end(),
-                      [](const scipp::index s) { return s == 1; })) {
-        throw except::BucketError("Not enough bin edges in dim " +
-                                  std::to_string(i) + ". Need at least 2.");
-      }
-    }
+  if (edges.empty() && groups.empty())
+    throw except::BucketError("Arguments 'edges' and 'groups' of scipp.bin are "
+                              "both empty. At least one must be set.");
+  for (const auto &edge : edges) {
+    const auto dim = edge.dims().inner();
```

Comment:
There is a prior call to a helper function `validate_bin_args` that does the validation. Also, `inner()` returns `Dim::Invalid` if it is 0-D, which would cause exceptions down the line, so this should be safe regardless.